// @aws-sdk/client-kendra/dist-es/commands/UpdateQuerySuggestionsConfigCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateQuerySuggestionsConfigRequestFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1UpdateQuerySuggestionsConfigCommand, serializeAws_json1_1UpdateQuerySuggestionsConfigCommand, } from "../protocols/Aws_json1_1";
export class UpdateQuerySuggestionsConfigCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, UpdateQuerySuggestionsConfigCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "KendraClient";
        const commandName = "UpdateQuerySuggestionsConfigCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateQuerySuggestionsConfigRequestFilterSensitiveLog,
            outputFilterSensitiveLog: (output) => output,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1UpdateQuerySuggestionsConfigCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1UpdateQuerySuggestionsConfigCommand(output, context);
    }
}
