// @aws-sdk/client-kendra/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { KendraServiceException as __BaseException } from "./KendraServiceException";
export class AccessDeniedException extends __BaseException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
export var HighlightType;
(function (HighlightType) {
    HighlightType["STANDARD"] = "STANDARD";
    HighlightType["THESAURUS_SYNONYM"] = "THESAURUS_SYNONYM";
})(HighlightType || (HighlightType = {}));
export var AdditionalResultAttributeValueType;
(function (AdditionalResultAttributeValueType) {
    AdditionalResultAttributeValueType["TEXT_WITH_HIGHLIGHTS_VALUE"] = "TEXT_WITH_HIGHLIGHTS_VALUE";
})(AdditionalResultAttributeValueType || (AdditionalResultAttributeValueType = {}));
export var AlfrescoEntity;
(function (AlfrescoEntity) {
    AlfrescoEntity["blog"] = "blog";
    AlfrescoEntity["documentLibrary"] = "documentLibrary";
    AlfrescoEntity["wiki"] = "wiki";
})(AlfrescoEntity || (AlfrescoEntity = {}));
export var EntityType;
(function (EntityType) {
    EntityType["GROUP"] = "GROUP";
    EntityType["USER"] = "USER";
})(EntityType || (EntityType = {}));
export class InternalServerException extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceAlreadyExistException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceAlreadyExistException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceAlreadyExistException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceNotFoundException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
export class ThrottlingException extends __BaseException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
    }
}
export class ValidationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
export var Persona;
(function (Persona) {
    Persona["OWNER"] = "OWNER";
    Persona["VIEWER"] = "VIEWER";
})(Persona || (Persona = {}));
export var ErrorCode;
(function (ErrorCode) {
    ErrorCode["INTERNAL_ERROR"] = "InternalError";
    ErrorCode["INVALID_REQUEST"] = "InvalidRequest";
})(ErrorCode || (ErrorCode = {}));
export class ConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
export var DocumentStatus;
(function (DocumentStatus) {
    DocumentStatus["FAILED"] = "FAILED";
    DocumentStatus["INDEXED"] = "INDEXED";
    DocumentStatus["NOT_FOUND"] = "NOT_FOUND";
    DocumentStatus["PROCESSING"] = "PROCESSING";
    DocumentStatus["UPDATED"] = "UPDATED";
    DocumentStatus["UPDATE_FAILED"] = "UPDATE_FAILED";
})(DocumentStatus || (DocumentStatus = {}));
export var ConditionOperator;
(function (ConditionOperator) {
    ConditionOperator["BeginsWith"] = "BeginsWith";
    ConditionOperator["Contains"] = "Contains";
    ConditionOperator["Equals"] = "Equals";
    ConditionOperator["Exists"] = "Exists";
    ConditionOperator["GreaterThan"] = "GreaterThan";
    ConditionOperator["GreaterThanOrEquals"] = "GreaterThanOrEquals";
    ConditionOperator["LessThan"] = "LessThan";
    ConditionOperator["LessThanOrEquals"] = "LessThanOrEquals";
    ConditionOperator["NotContains"] = "NotContains";
    ConditionOperator["NotEquals"] = "NotEquals";
    ConditionOperator["NotExists"] = "NotExists";
})(ConditionOperator || (ConditionOperator = {}));
export var ReadAccessType;
(function (ReadAccessType) {
    ReadAccessType["ALLOW"] = "ALLOW";
    ReadAccessType["DENY"] = "DENY";
})(ReadAccessType || (ReadAccessType = {}));
export var PrincipalType;
(function (PrincipalType) {
    PrincipalType["GROUP"] = "GROUP";
    PrincipalType["USER"] = "USER";
})(PrincipalType || (PrincipalType = {}));
export var ContentType;
(function (ContentType) {
    ContentType["CSV"] = "CSV";
    ContentType["HTML"] = "HTML";
    ContentType["JSON"] = "JSON";
    ContentType["MD"] = "MD";
    ContentType["MS_EXCEL"] = "MS_EXCEL";
    ContentType["MS_WORD"] = "MS_WORD";
    ContentType["PDF"] = "PDF";
    ContentType["PLAIN_TEXT"] = "PLAIN_TEXT";
    ContentType["PPT"] = "PPT";
    ContentType["RTF"] = "RTF";
    ContentType["XML"] = "XML";
    ContentType["XSLT"] = "XSLT";
})(ContentType || (ContentType = {}));
export class ServiceQuotaExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
export var ConfluenceAttachmentFieldName;
(function (ConfluenceAttachmentFieldName) {
    ConfluenceAttachmentFieldName["AUTHOR"] = "AUTHOR";
    ConfluenceAttachmentFieldName["CONTENT_TYPE"] = "CONTENT_TYPE";
    ConfluenceAttachmentFieldName["CREATED_DATE"] = "CREATED_DATE";
    ConfluenceAttachmentFieldName["DISPLAY_URL"] = "DISPLAY_URL";
    ConfluenceAttachmentFieldName["FILE_SIZE"] = "FILE_SIZE";
    ConfluenceAttachmentFieldName["ITEM_TYPE"] = "ITEM_TYPE";
    ConfluenceAttachmentFieldName["PARENT_ID"] = "PARENT_ID";
    ConfluenceAttachmentFieldName["SPACE_KEY"] = "SPACE_KEY";
    ConfluenceAttachmentFieldName["SPACE_NAME"] = "SPACE_NAME";
    ConfluenceAttachmentFieldName["URL"] = "URL";
    ConfluenceAttachmentFieldName["VERSION"] = "VERSION";
})(ConfluenceAttachmentFieldName || (ConfluenceAttachmentFieldName = {}));
export var ConfluenceAuthenticationType;
(function (ConfluenceAuthenticationType) {
    ConfluenceAuthenticationType["HTTP_BASIC"] = "HTTP_BASIC";
    ConfluenceAuthenticationType["PAT"] = "PAT";
})(ConfluenceAuthenticationType || (ConfluenceAuthenticationType = {}));
export var ConfluenceBlogFieldName;
(function (ConfluenceBlogFieldName) {
    ConfluenceBlogFieldName["AUTHOR"] = "AUTHOR";
    ConfluenceBlogFieldName["DISPLAY_URL"] = "DISPLAY_URL";
    ConfluenceBlogFieldName["ITEM_TYPE"] = "ITEM_TYPE";
    ConfluenceBlogFieldName["LABELS"] = "LABELS";
    ConfluenceBlogFieldName["PUBLISH_DATE"] = "PUBLISH_DATE";
    ConfluenceBlogFieldName["SPACE_KEY"] = "SPACE_KEY";
    ConfluenceBlogFieldName["SPACE_NAME"] = "SPACE_NAME";
    ConfluenceBlogFieldName["URL"] = "URL";
    ConfluenceBlogFieldName["VERSION"] = "VERSION";
})(ConfluenceBlogFieldName || (ConfluenceBlogFieldName = {}));
export var ConfluencePageFieldName;
(function (ConfluencePageFieldName) {
    ConfluencePageFieldName["AUTHOR"] = "AUTHOR";
    ConfluencePageFieldName["CONTENT_STATUS"] = "CONTENT_STATUS";
    ConfluencePageFieldName["CREATED_DATE"] = "CREATED_DATE";
    ConfluencePageFieldName["DISPLAY_URL"] = "DISPLAY_URL";
    ConfluencePageFieldName["ITEM_TYPE"] = "ITEM_TYPE";
    ConfluencePageFieldName["LABELS"] = "LABELS";
    ConfluencePageFieldName["MODIFIED_DATE"] = "MODIFIED_DATE";
    ConfluencePageFieldName["PARENT_ID"] = "PARENT_ID";
    ConfluencePageFieldName["SPACE_KEY"] = "SPACE_KEY";
    ConfluencePageFieldName["SPACE_NAME"] = "SPACE_NAME";
    ConfluencePageFieldName["URL"] = "URL";
    ConfluencePageFieldName["VERSION"] = "VERSION";
})(ConfluencePageFieldName || (ConfluencePageFieldName = {}));
export var ConfluenceSpaceFieldName;
(function (ConfluenceSpaceFieldName) {
    ConfluenceSpaceFieldName["DISPLAY_URL"] = "DISPLAY_URL";
    ConfluenceSpaceFieldName["ITEM_TYPE"] = "ITEM_TYPE";
    ConfluenceSpaceFieldName["SPACE_KEY"] = "SPACE_KEY";
    ConfluenceSpaceFieldName["URL"] = "URL";
})(ConfluenceSpaceFieldName || (ConfluenceSpaceFieldName = {}));
export var ConfluenceVersion;
(function (ConfluenceVersion) {
    ConfluenceVersion["CLOUD"] = "CLOUD";
    ConfluenceVersion["SERVER"] = "SERVER";
})(ConfluenceVersion || (ConfluenceVersion = {}));
export var DatabaseEngineType;
(function (DatabaseEngineType) {
    DatabaseEngineType["RDS_AURORA_MYSQL"] = "RDS_AURORA_MYSQL";
    DatabaseEngineType["RDS_AURORA_POSTGRESQL"] = "RDS_AURORA_POSTGRESQL";
    DatabaseEngineType["RDS_MYSQL"] = "RDS_MYSQL";
    DatabaseEngineType["RDS_POSTGRESQL"] = "RDS_POSTGRESQL";
})(DatabaseEngineType || (DatabaseEngineType = {}));
export var QueryIdentifiersEnclosingOption;
(function (QueryIdentifiersEnclosingOption) {
    QueryIdentifiersEnclosingOption["DOUBLE_QUOTES"] = "DOUBLE_QUOTES";
    QueryIdentifiersEnclosingOption["NONE"] = "NONE";
})(QueryIdentifiersEnclosingOption || (QueryIdentifiersEnclosingOption = {}));
export var FsxFileSystemType;
(function (FsxFileSystemType) {
    FsxFileSystemType["WINDOWS"] = "WINDOWS";
})(FsxFileSystemType || (FsxFileSystemType = {}));
export var Type;
(function (Type) {
    Type["ON_PREMISE"] = "ON_PREMISE";
    Type["SAAS"] = "SAAS";
})(Type || (Type = {}));
export var IssueSubEntity;
(function (IssueSubEntity) {
    IssueSubEntity["ATTACHMENTS"] = "ATTACHMENTS";
    IssueSubEntity["COMMENTS"] = "COMMENTS";
    IssueSubEntity["WORKLOGS"] = "WORKLOGS";
})(IssueSubEntity || (IssueSubEntity = {}));
export var SalesforceChatterFeedIncludeFilterType;
(function (SalesforceChatterFeedIncludeFilterType) {
    SalesforceChatterFeedIncludeFilterType["ACTIVE_USER"] = "ACTIVE_USER";
    SalesforceChatterFeedIncludeFilterType["STANDARD_USER"] = "STANDARD_USER";
})(SalesforceChatterFeedIncludeFilterType || (SalesforceChatterFeedIncludeFilterType = {}));
export var SalesforceKnowledgeArticleState;
(function (SalesforceKnowledgeArticleState) {
    SalesforceKnowledgeArticleState["ARCHIVED"] = "ARCHIVED";
    SalesforceKnowledgeArticleState["DRAFT"] = "DRAFT";
    SalesforceKnowledgeArticleState["PUBLISHED"] = "PUBLISHED";
})(SalesforceKnowledgeArticleState || (SalesforceKnowledgeArticleState = {}));
export var SalesforceStandardObjectName;
(function (SalesforceStandardObjectName) {
    SalesforceStandardObjectName["ACCOUNT"] = "ACCOUNT";
    SalesforceStandardObjectName["CAMPAIGN"] = "CAMPAIGN";
    SalesforceStandardObjectName["CASE"] = "CASE";
    SalesforceStandardObjectName["CONTACT"] = "CONTACT";
    SalesforceStandardObjectName["CONTRACT"] = "CONTRACT";
    SalesforceStandardObjectName["DOCUMENT"] = "DOCUMENT";
    SalesforceStandardObjectName["GROUP"] = "GROUP";
    SalesforceStandardObjectName["IDEA"] = "IDEA";
    SalesforceStandardObjectName["LEAD"] = "LEAD";
    SalesforceStandardObjectName["OPPORTUNITY"] = "OPPORTUNITY";
    SalesforceStandardObjectName["PARTNER"] = "PARTNER";
    SalesforceStandardObjectName["PRICEBOOK"] = "PRICEBOOK";
    SalesforceStandardObjectName["PRODUCT"] = "PRODUCT";
    SalesforceStandardObjectName["PROFILE"] = "PROFILE";
    SalesforceStandardObjectName["SOLUTION"] = "SOLUTION";
    SalesforceStandardObjectName["TASK"] = "TASK";
    SalesforceStandardObjectName["USER"] = "USER";
})(SalesforceStandardObjectName || (SalesforceStandardObjectName = {}));
export var ServiceNowAuthenticationType;
(function (ServiceNowAuthenticationType) {
    ServiceNowAuthenticationType["HTTP_BASIC"] = "HTTP_BASIC";
    ServiceNowAuthenticationType["OAUTH2"] = "OAUTH2";
})(ServiceNowAuthenticationType || (ServiceNowAuthenticationType = {}));
export var ServiceNowBuildVersionType;
(function (ServiceNowBuildVersionType) {
    ServiceNowBuildVersionType["LONDON"] = "LONDON";
    ServiceNowBuildVersionType["OTHERS"] = "OTHERS";
})(ServiceNowBuildVersionType || (ServiceNowBuildVersionType = {}));
export var SharePointOnlineAuthenticationType;
(function (SharePointOnlineAuthenticationType) {
    SharePointOnlineAuthenticationType["HTTP_BASIC"] = "HTTP_BASIC";
    SharePointOnlineAuthenticationType["OAUTH2"] = "OAUTH2";
})(SharePointOnlineAuthenticationType || (SharePointOnlineAuthenticationType = {}));
export var SharePointVersion;
(function (SharePointVersion) {
    SharePointVersion["SHAREPOINT_2013"] = "SHAREPOINT_2013";
    SharePointVersion["SHAREPOINT_2016"] = "SHAREPOINT_2016";
    SharePointVersion["SHAREPOINT_2019"] = "SHAREPOINT_2019";
    SharePointVersion["SHAREPOINT_ONLINE"] = "SHAREPOINT_ONLINE";
})(SharePointVersion || (SharePointVersion = {}));
export var SlackEntity;
(function (SlackEntity) {
    SlackEntity["DIRECT_MESSAGE"] = "DIRECT_MESSAGE";
    SlackEntity["GROUP_MESSAGE"] = "GROUP_MESSAGE";
    SlackEntity["PRIVATE_CHANNEL"] = "PRIVATE_CHANNEL";
    SlackEntity["PUBLIC_CHANNEL"] = "PUBLIC_CHANNEL";
})(SlackEntity || (SlackEntity = {}));
export var WebCrawlerMode;
(function (WebCrawlerMode) {
    WebCrawlerMode["EVERYTHING"] = "EVERYTHING";
    WebCrawlerMode["HOST_ONLY"] = "HOST_ONLY";
    WebCrawlerMode["SUBDOMAINS"] = "SUBDOMAINS";
})(WebCrawlerMode || (WebCrawlerMode = {}));
export var DataSourceType;
(function (DataSourceType) {
    DataSourceType["ALFRESCO"] = "ALFRESCO";
    DataSourceType["BOX"] = "BOX";
    DataSourceType["CONFLUENCE"] = "CONFLUENCE";
    DataSourceType["CUSTOM"] = "CUSTOM";
    DataSourceType["DATABASE"] = "DATABASE";
    DataSourceType["FSX"] = "FSX";
    DataSourceType["GITHUB"] = "GITHUB";
    DataSourceType["GOOGLEDRIVE"] = "GOOGLEDRIVE";
    DataSourceType["JIRA"] = "JIRA";
    DataSourceType["ONEDRIVE"] = "ONEDRIVE";
    DataSourceType["QUIP"] = "QUIP";
    DataSourceType["S3"] = "S3";
    DataSourceType["SALESFORCE"] = "SALESFORCE";
    DataSourceType["SERVICENOW"] = "SERVICENOW";
    DataSourceType["SHAREPOINT"] = "SHAREPOINT";
    DataSourceType["SLACK"] = "SLACK";
    DataSourceType["TEMPLATE"] = "TEMPLATE";
    DataSourceType["WEBCRAWLER"] = "WEBCRAWLER";
    DataSourceType["WORKDOCS"] = "WORKDOCS";
})(DataSourceType || (DataSourceType = {}));
export var FaqFileFormat;
(function (FaqFileFormat) {
    FaqFileFormat["CSV"] = "CSV";
    FaqFileFormat["CSV_WITH_HEADER"] = "CSV_WITH_HEADER";
    FaqFileFormat["JSON"] = "JSON";
})(FaqFileFormat || (FaqFileFormat = {}));
export var IndexEdition;
(function (IndexEdition) {
    IndexEdition["DEVELOPER_EDITION"] = "DEVELOPER_EDITION";
    IndexEdition["ENTERPRISE_EDITION"] = "ENTERPRISE_EDITION";
})(IndexEdition || (IndexEdition = {}));
export var UserContextPolicy;
(function (UserContextPolicy) {
    UserContextPolicy["ATTRIBUTE_FILTER"] = "ATTRIBUTE_FILTER";
    UserContextPolicy["USER_TOKEN"] = "USER_TOKEN";
})(UserContextPolicy || (UserContextPolicy = {}));
export var UserGroupResolutionMode;
(function (UserGroupResolutionMode) {
    UserGroupResolutionMode["AWS_SSO"] = "AWS_SSO";
    UserGroupResolutionMode["NONE"] = "NONE";
})(UserGroupResolutionMode || (UserGroupResolutionMode = {}));
export var KeyLocation;
(function (KeyLocation) {
    KeyLocation["SECRET_MANAGER"] = "SECRET_MANAGER";
    KeyLocation["URL"] = "URL";
})(KeyLocation || (KeyLocation = {}));
export var DataSourceStatus;
(function (DataSourceStatus) {
    DataSourceStatus["ACTIVE"] = "ACTIVE";
    DataSourceStatus["CREATING"] = "CREATING";
    DataSourceStatus["DELETING"] = "DELETING";
    DataSourceStatus["FAILED"] = "FAILED";
    DataSourceStatus["UPDATING"] = "UPDATING";
})(DataSourceStatus || (DataSourceStatus = {}));
export var EndpointType;
(function (EndpointType) {
    EndpointType["HOME"] = "HOME";
})(EndpointType || (EndpointType = {}));
export var ExperienceStatus;
(function (ExperienceStatus) {
    ExperienceStatus["ACTIVE"] = "ACTIVE";
    ExperienceStatus["CREATING"] = "CREATING";
    ExperienceStatus["DELETING"] = "DELETING";
    ExperienceStatus["FAILED"] = "FAILED";
})(ExperienceStatus || (ExperienceStatus = {}));
export var FaqStatus;
(function (FaqStatus) {
    FaqStatus["ACTIVE"] = "ACTIVE";
    FaqStatus["CREATING"] = "CREATING";
    FaqStatus["DELETING"] = "DELETING";
    FaqStatus["FAILED"] = "FAILED";
    FaqStatus["UPDATING"] = "UPDATING";
})(FaqStatus || (FaqStatus = {}));
export var Order;
(function (Order) {
    Order["ASCENDING"] = "ASCENDING";
    Order["DESCENDING"] = "DESCENDING";
})(Order || (Order = {}));
export var DocumentAttributeValueType;
(function (DocumentAttributeValueType) {
    DocumentAttributeValueType["DATE_VALUE"] = "DATE_VALUE";
    DocumentAttributeValueType["LONG_VALUE"] = "LONG_VALUE";
    DocumentAttributeValueType["STRING_LIST_VALUE"] = "STRING_LIST_VALUE";
    DocumentAttributeValueType["STRING_VALUE"] = "STRING_VALUE";
})(DocumentAttributeValueType || (DocumentAttributeValueType = {}));
export var IndexStatus;
(function (IndexStatus) {
    IndexStatus["ACTIVE"] = "ACTIVE";
    IndexStatus["CREATING"] = "CREATING";
    IndexStatus["DELETING"] = "DELETING";
    IndexStatus["FAILED"] = "FAILED";
    IndexStatus["SYSTEM_UPDATING"] = "SYSTEM_UPDATING";
    IndexStatus["UPDATING"] = "UPDATING";
})(IndexStatus || (IndexStatus = {}));
export var PrincipalMappingStatus;
(function (PrincipalMappingStatus) {
    PrincipalMappingStatus["DELETED"] = "DELETED";
    PrincipalMappingStatus["DELETING"] = "DELETING";
    PrincipalMappingStatus["FAILED"] = "FAILED";
    PrincipalMappingStatus["PROCESSING"] = "PROCESSING";
    PrincipalMappingStatus["SUCCEEDED"] = "SUCCEEDED";
})(PrincipalMappingStatus || (PrincipalMappingStatus = {}));
export var QuerySuggestionsBlockListStatus;
(function (QuerySuggestionsBlockListStatus) {
    QuerySuggestionsBlockListStatus["ACTIVE"] = "ACTIVE";
    QuerySuggestionsBlockListStatus["ACTIVE_BUT_UPDATE_FAILED"] = "ACTIVE_BUT_UPDATE_FAILED";
    QuerySuggestionsBlockListStatus["CREATING"] = "CREATING";
    QuerySuggestionsBlockListStatus["DELETING"] = "DELETING";
    QuerySuggestionsBlockListStatus["FAILED"] = "FAILED";
    QuerySuggestionsBlockListStatus["UPDATING"] = "UPDATING";
})(QuerySuggestionsBlockListStatus || (QuerySuggestionsBlockListStatus = {}));
export var Mode;
(function (Mode) {
    Mode["ENABLED"] = "ENABLED";
    Mode["LEARN_ONLY"] = "LEARN_ONLY";
})(Mode || (Mode = {}));
export var QuerySuggestionsStatus;
(function (QuerySuggestionsStatus) {
    QuerySuggestionsStatus["ACTIVE"] = "ACTIVE";
    QuerySuggestionsStatus["UPDATING"] = "UPDATING";
})(QuerySuggestionsStatus || (QuerySuggestionsStatus = {}));
export var ThesaurusStatus;
(function (ThesaurusStatus) {
    ThesaurusStatus["ACTIVE"] = "ACTIVE";
    ThesaurusStatus["ACTIVE_BUT_UPDATE_FAILED"] = "ACTIVE_BUT_UPDATE_FAILED";
    ThesaurusStatus["CREATING"] = "CREATING";
    ThesaurusStatus["DELETING"] = "DELETING";
    ThesaurusStatus["FAILED"] = "FAILED";
    ThesaurusStatus["UPDATING"] = "UPDATING";
})(ThesaurusStatus || (ThesaurusStatus = {}));
export var Interval;
(function (Interval) {
    Interval["ONE_MONTH_AGO"] = "ONE_MONTH_AGO";
    Interval["ONE_WEEK_AGO"] = "ONE_WEEK_AGO";
    Interval["THIS_MONTH"] = "THIS_MONTH";
    Interval["THIS_WEEK"] = "THIS_WEEK";
    Interval["TWO_MONTHS_AGO"] = "TWO_MONTHS_AGO";
    Interval["TWO_WEEKS_AGO"] = "TWO_WEEKS_AGO";
})(Interval || (Interval = {}));
export var MetricType;
(function (MetricType) {
    MetricType["AGG_QUERY_DOC_METRICS"] = "AGG_QUERY_DOC_METRICS";
    MetricType["DOCS_BY_CLICK_COUNT"] = "DOCS_BY_CLICK_COUNT";
    MetricType["QUERIES_BY_COUNT"] = "QUERIES_BY_COUNT";
    MetricType["QUERIES_BY_ZERO_CLICK_RATE"] = "QUERIES_BY_ZERO_CLICK_RATE";
    MetricType["QUERIES_BY_ZERO_RESULT_RATE"] = "QUERIES_BY_ZERO_RESULT_RATE";
    MetricType["TREND_QUERY_DOC_METRICS"] = "TREND_QUERY_DOC_METRICS";
})(MetricType || (MetricType = {}));
export class InvalidRequestException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.Message = opts.Message;
    }
}
export var DataSourceSyncJobStatus;
(function (DataSourceSyncJobStatus) {
    DataSourceSyncJobStatus["ABORTED"] = "ABORTED";
    DataSourceSyncJobStatus["FAILED"] = "FAILED";
    DataSourceSyncJobStatus["INCOMPLETE"] = "INCOMPLETE";
    DataSourceSyncJobStatus["STOPPING"] = "STOPPING";
    DataSourceSyncJobStatus["SUCCEEDED"] = "SUCCEEDED";
    DataSourceSyncJobStatus["SYNCING"] = "SYNCING";
    DataSourceSyncJobStatus["SYNCING_INDEXING"] = "SYNCING_INDEXING";
})(DataSourceSyncJobStatus || (DataSourceSyncJobStatus = {}));
export class ResourceUnavailableException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceUnavailableException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceUnavailableException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceUnavailableException.prototype);
        this.Message = opts.Message;
    }
}
export var QueryResultType;
(function (QueryResultType) {
    QueryResultType["ANSWER"] = "ANSWER";
    QueryResultType["DOCUMENT"] = "DOCUMENT";
    QueryResultType["QUESTION_ANSWER"] = "QUESTION_ANSWER";
})(QueryResultType || (QueryResultType = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var QueryResultFormat;
(function (QueryResultFormat) {
    QueryResultFormat["TABLE"] = "TABLE";
    QueryResultFormat["TEXT"] = "TEXT";
})(QueryResultFormat || (QueryResultFormat = {}));
export var ScoreConfidence;
(function (ScoreConfidence) {
    ScoreConfidence["HIGH"] = "HIGH";
    ScoreConfidence["LOW"] = "LOW";
    ScoreConfidence["MEDIUM"] = "MEDIUM";
    ScoreConfidence["NOT_AVAILABLE"] = "NOT_AVAILABLE";
    ScoreConfidence["VERY_HIGH"] = "VERY_HIGH";
})(ScoreConfidence || (ScoreConfidence = {}));
export var WarningCode;
(function (WarningCode) {
    WarningCode["QUERY_LANGUAGE_INVALID_SYNTAX"] = "QUERY_LANGUAGE_INVALID_SYNTAX";
})(WarningCode || (WarningCode = {}));
export class ResourceInUseException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceInUseException.prototype);
        this.Message = opts.Message;
    }
}
export var RelevanceType;
(function (RelevanceType) {
    RelevanceType["NOT_RELEVANT"] = "NOT_RELEVANT";
    RelevanceType["RELEVANT"] = "RELEVANT";
})(RelevanceType || (RelevanceType = {}));
export const AccessControlConfigurationSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AccessControlListConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AclConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const HighlightFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TextWithHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AdditionalResultAttributeValueFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AdditionalResultAttributeFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceToIndexFieldMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const S3PathFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceVpcConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AlfrescoConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EntityConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AssociateEntitiesToExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FailedEntityFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AssociateEntitiesToExperienceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EntityPersonaConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AssociatePersonasToEntitiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AssociatePersonasToEntitiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentAttributeValueFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentAttributeFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BasicAuthenticationConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AuthenticationConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceSyncJobMetricTargetFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchDeleteDocumentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchDeleteDocumentResponseFailedDocumentFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchDeleteDocumentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchGetDocumentStatusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StatusFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchGetDocumentStatusResponseErrorFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchGetDocumentStatusResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentAttributeConditionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentAttributeTargetFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const InlineCustomDocumentEnrichmentConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const HookConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CustomDocumentEnrichmentConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PrincipalFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const HierarchicalPrincipalFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchPutDocumentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchPutDocumentResponseFailedDocumentFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BatchPutDocumentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ClearQuerySuggestionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateAccessControlConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateAccessControlConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const BoxConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceAttachmentToIndexFieldMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceAttachmentConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceBlogToIndexFieldMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceBlogConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluencePageToIndexFieldMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluencePageConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ProxyConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceSpaceToIndexFieldMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceSpaceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConfluenceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ColumnConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ConnectionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SqlConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DatabaseConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FsxConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GitHubDocumentCrawlPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OnPremiseConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SaaSConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GitHubConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GoogleDriveConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const JiraConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OneDriveUsersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OneDriveConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QuipConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentsMetadataConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const S3DataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceChatterFeedConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceCustomKnowledgeArticleTypeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceStandardKnowledgeArticleTypeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceKnowledgeArticleConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceStandardObjectAttachmentConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceStandardObjectConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SalesforceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ServiceNowKnowledgeArticleConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ServiceNowServiceCatalogConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ServiceNowConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SharePointConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SlackConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TemplateConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SeedUrlConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SiteMapsConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UrlsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const WebCrawlerConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const WorkDocsConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateDataSourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContentSourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UserIdentityConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ExperienceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateExperienceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFaqRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFaqResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ServerSideEncryptionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.KmsKeyId && { KmsKeyId: SENSITIVE_STRING }),
});
export const UserGroupResolutionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const JsonTokenTypeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const JwtTokenTypeConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UserTokenConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateIndexRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ServerSideEncryptionConfiguration && {
        ServerSideEncryptionConfiguration: ServerSideEncryptionConfigurationFilterSensitiveLog(obj.ServerSideEncryptionConfiguration),
    }),
});
export const CreateIndexResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateQuerySuggestionsBlockListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateQuerySuggestionsBlockListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateThesaurusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateThesaurusResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteAccessControlConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteAccessControlConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteExperienceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteFaqRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteIndexRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeletePrincipalMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteQuerySuggestionsBlockListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteThesaurusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeAccessControlConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeAccessControlConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeDataSourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ExperienceEndpointFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeExperienceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeFaqRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeFaqResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeIndexRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CapacityUnitsConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RelevanceFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SearchFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentMetadataConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FaqStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TextDocumentStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const IndexStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeIndexResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ServerSideEncryptionConfiguration && {
        ServerSideEncryptionConfiguration: ServerSideEncryptionConfigurationFilterSensitiveLog(obj.ServerSideEncryptionConfiguration),
    }),
});
export const DescribePrincipalMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GroupOrderingIdSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribePrincipalMappingResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeQuerySuggestionsBlockListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeQuerySuggestionsBlockListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeQuerySuggestionsConfigRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeQuerySuggestionsConfigResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeThesaurusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeThesaurusResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DisassociateEntitiesFromExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DisassociateEntitiesFromExperienceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DisassociatePersonasFromEntitiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DisassociatePersonasFromEntitiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetQuerySuggestionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SuggestionHighlightFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SuggestionTextWithHighlightsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SuggestionValueFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SuggestionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetQuerySuggestionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetSnapshotsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TimeRangeFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetSnapshotsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListAccessControlConfigurationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListAccessControlConfigurationsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListDataSourcesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListDataSourcesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListDataSourceSyncJobsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceSyncJobMetricsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceSyncJobFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListDataSourceSyncJobsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListEntityPersonasRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PersonasSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListEntityPersonasResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListExperienceEntitiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EntityDisplayDataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.UserName && { UserName: SENSITIVE_STRING }),
    ...(obj.GroupName && { GroupName: SENSITIVE_STRING }),
    ...(obj.IdentifiedUserName && { IdentifiedUserName: SENSITIVE_STRING }),
    ...(obj.FirstName && { FirstName: SENSITIVE_STRING }),
    ...(obj.LastName && { LastName: SENSITIVE_STRING }),
});
export const ExperienceEntitiesSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DisplayData && { DisplayData: EntityDisplayDataFilterSensitiveLog(obj.DisplayData) }),
});
export const ListExperienceEntitiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.SummaryItems && {
        SummaryItems: obj.SummaryItems.map((item) => ExperienceEntitiesSummaryFilterSensitiveLog(item)),
    }),
});
export const ListExperiencesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ExperiencesSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListExperiencesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListFaqsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FaqSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListFaqsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListGroupsOlderThanOrderingIdRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GroupSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListGroupsOlderThanOrderingIdResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListIndicesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const IndexConfigurationSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListIndicesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListQuerySuggestionsBlockListsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QuerySuggestionsBlockListSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListQuerySuggestionsBlockListsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListTagsForResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListTagsForResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListThesauriRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ThesaurusSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListThesauriResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MemberGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MemberUserFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GroupMembersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PutPrincipalMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentRelevanceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SortingConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SpellCorrectionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DataSourceGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UserContextFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ScoreAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TableCellFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TableRowFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TableExcerptFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryResultItemFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CorrectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SpellCorrectedQueryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const WarningFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StartDataSourceSyncJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StartDataSourceSyncJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StopDataSourceSyncJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ClickFeedbackFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RelevanceFeedbackFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SubmitFeedbackRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateAccessControlConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateAccessControlConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateDataSourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateExperienceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateIndexRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateQuerySuggestionsBlockListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
